src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/VmrDiffOperation.cs

```diff
@@ -149,26 +148,63 @@ await processManager.ExecuteGit(Path.GetDirectoryName(tmpProductRepo)!, [
     private async Task<IReadOnlyCollection<string>> GetDiffFilters(string mapping)
     {
         var gitRepo = gitRepoFactory.CreateClient(DarcLib.Constants.DefaultVmrUri);
-        var repo = await remoteFactory.CreateRemoteAsync(DarcLib.Constants.DefaultVmrUri);
         return sourceMappingParser.ParseMappingsFromJson(
             (await gitRepo.GetFileContentsAsync($"{VmrInfo.SourceDirName}/{VmrInfo.SourceMappingsFileName}", DarcLib.Constants.DefaultVmrUri, "main")))
             .First(m => m.Name == mapping).Exclude;
     }
 
+    private async Task<string> GetDefaultVmrRepoRef(string defaultRemote)
+    {
+        var gitRepo = gitRepoFactory.CreateClient(DarcLib.Constants.DefaultVmrUri);
+        return sourceMappingParser.ParseMappingsFromJson(
+            (await gitRepo.GetFileContentsAsync($"{VmrInfo.SourceDirName}/{VmrInfo.SourceMappingsFileName}", DarcLib.Constants.DefaultVmrUri, "main")))
+            .First(m => m.DefaultRemote == defaultRemote).DefaultRef;
```

Comment:
Can we break this apart a bit?

Reply:
I made most of this a part of an IRemote method instead